src/libraries/Common/src/Interop/Unix/System.Native/Interop.ReadLink.cs

```diff
@@ -35,26 +35,35 @@ internal static partial class Sys
             // Use an initial buffer size that prevents disposing and renting
             // a second time when calling ConvertAndTerminateString.
             using var converter = new ValueUtf8Converter(stackalloc byte[1024]);
+            Span<byte> stackBuffer = stackalloc byte[outputBufferSize];
```

Comment:
This and the line above puts 2K on the stack - we normally don't seem to stackalloc more than 512 bytes as a rule of thumb.

Reply:
@carlossanlop why is ValueUtf8Converter using 1024 here? all other callsites use `DefaultPathBufferSize`, that is 256. 
https://github.com/dotnet/runtime/blob/c2ec86b1c552ac8a1749f9f98e012f707e325660/src/libraries/Common/src/Interop/Unix/System.Native/Interop.Stat.Span.cs#L17
https://github.com/dotnet/runtime/blob/c2ec86b1c552ac8a1749f9f98e012f707e325660/src/libraries/Common/src/Interop/Unix/System.Native/Interop.Rename.cs#L28-L29
Perhaps we can switch to `new ValueUtf8Converter(DefaultPathBufferSize)` and `outputBuffersize = 256` so we can keep it on the 512 threshold.

Reply:
I had originally proposed using 256 as the initial value https://github.com/dotnet/runtime/pull/54253#discussion_r658578727  and then I got the suggestion to just go ahead and pre-reserve 1024 from the beginning. The argument was that _renting_ is low cost for us.

Considering this change is now doing a `stackalloc`, the initial value used for that should be lower, based on what @danmoseley mentioned.

In Unix, the max _filename length_ is 256, and the max _path length_ is 4096.

@pedrobsaila I propose we use 256 as the `stackalloc` buffer size since it's the max filename limit.

Reply:
`1024` is often the default because this is the official threshold for `_malloca` on Windows. `_malloca` is an API that allocates on the stack if `size` is below the threshold and allocates on the native heap (calls `malloc`) otherwise.

The threshold is defined by `_ALLOCA_S_THRESHOLD`.

If you need less, its definitely better to do so. Likewise there may be considerations for doing stack allocation in loops or non-leaf methods if there will be further calls that themselves do heavy stack allocation. But, `1024` is a good upper limit and a reasonable bounds otherwise.

Reply:
> But, 1024 is a good upper limit and a reasonable bounds otherwise.

I have no objection, but we should be consistent. We seem to generally use no more than 512 bytes. Maybe we should have a higher limit where we're confident it's a leaf method like this. cc @stephentoub

Reply:
These are paths. When was it last time you have seen a path longer than 512 chars (outside dotnet/runtime tests)?

Reply:
> We seem to generally use no more than 512 bytes

We vary, but we frequently go up to 1K.  It should also be super rare to get anywhere close to these numbers.